apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/AbstractCleanupJob.java

```diff
@@ -109,48 +105,42 @@ public int run() throws ProcessingException {
    * Executes a single delete transaction. The list of queries are passed in execution order, last
    * query must delete from the parent table, all other queries are child table deletes.
    *
-   * @param queries the queries to execute in each transaction.
+   * @param query the queries to execute in each transaction.
    * @param tm the TransactionManager to use with the deletion.
    * @return the number of claims deleted by the transaction.
    */
-  private int executeDeleteTransaction(List<Query> queries, TransactionManager tm) {
+  private int executeDeleteTransaction(Query query, TransactionManager tm) {
     return tm.executeFunction(
         entityManager -> {
           // execute all queries in order, use the count from the last query (parent table)
           // to track the number of claims deleted.
           var count = 0;
-          for (Query query : queries) {
-            count = query.executeUpdate();
-          }
+          count = query.executeUpdate();
           return count;
         });
   }
 
   /**
    * Build a list of native sql delete queries to remove claims.
    *
-   * @param cutoffDate the Instant used to identify queries for deletion.
    * @param tm the TransactionManager to use to create queries.
    * @return the list of queries.
    */
-  private List<Query> buildDeleteQueries(Instant cutoffDate, TransactionManager tm) {
-    List<Query> queries = new ArrayList<>();
-    String parentTableName = getTableNames().getLast();
+  private Query buildDeleteQuery(TransactionManager tm) {
+    String parentTableName = getParentTableName();
+    AtomicReference<Query> atomicQuery = new AtomicReference<>();
     tm.executeProcedure(
```

Comment:
tm.getEntityManager() doesn't actually create the entity manager if it doesn't exist; exectuteProcedure does. This is the pattern that's used throughout the application, so sticking to it.

Reply:
Good to know, thanks for checking. Maybe `TransactionManager` should be extended to support this better, since it's a bit odd to start a database transaction without actually doing any queries, but that's probably outside the scope of this PR.